### PR TITLE
Polygon Forest LP Layers

### PR DESCRIPTION
**Pull request by ManuelStardust**

### CODE REVIEW

File: src/pages/Search.jsx [hunk 1]
```diff
@@ -611,12 +613,14 @@ class Search extends Component {
         layerStyle = this.featureStyle({ type: layerName });
         break;
       case 'geofence':
-        reqPromise = () => RestAPI.requestGeofenceGeometryByArea(
-          selectedAreaTypeId,
-          selectedAreaId,
-        );
-        if (styleName === "border") {
-          layerStyle = { stroke: true, color: matchColor("border")(), weight: 2, opacity: 1, fillOpacity: 0 };
+        if(searchType === "definedArea"){
```
Comment (erikasv) [inline src/pages/Search.jsx:616]: what happen if is not a defined area?

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts
Comment (erikasv) [inline src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:1]: Is not from this PR, but accessing the data prop without checking the object existence (line 86) leads to an unhandled error

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts [hunk 2]
```diff
@@ -56,11 +56,12 @@ export class ForestLossPersistenceController {
...
-    polygon: Polygon | null
+    polygon: Polygon | null,
+    polygonFolder: string
   ): Promise<ForestLPData> => {
     if (searchType === "drawPolygon") {
       return biabAPI
-        .requestForestLPData(polygon)
+        .requestForestLPData(polygon, polygonFolder)
         .then((data) => {
           const periods = ["2000-2005", "2006-2010", "2011-2015", "2016-2021"];
```
[the remark this answers is not in the document]
Comment (erikasv) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:66]: sorry, I don't know who did this change, so maybe @cazapatamar should be tagged xD

[the remark this answers is not in the document]
Comment (cazapatamar) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:66]: Yes, it is mine. I used the same logic as in the back to process the list of periods but you are right it could be improved as you say as it is done in defined area case.

@ManuelStardust As you prefer, if you consider you can make this change or i could fix it later once this PR is merged. Please LMN.

File: src/pages/Search.jsx [hunk 3]
```diff
@@ -746,12 +751,16 @@ class Search extends Component {
...
       const [, yearIni, yearEnd, category] = layerName.match(/forestLP-(\w+)-(\w+)-(\w+)/);
-      reqPromise = () => RestAPI.requestForestLPLayer(
-        selectedAreaTypeId,
-        selectedAreaId,
-        `${yearIni}-${yearEnd}`,
-        category,
-      );
+      if(searchType === "drawPolygon"){
+        reqPromise = () => biabAPI.requestForestLPLayer(layerName, polygonFolder);
+      }else{
```
[the remark this answers is not in the document]
Comment (ManuelStardust) [reply src/pages/Search.jsx]: Solved

File: src/utils/biabAPI.ts [hunk 4]
```diff
@@ -40,6 +43,31 @@ class biabAPI {
...
+   *
+   * @return {Promise<Object>} layer object to be loaded in the map
+   */
+
+  static requestForestLPLayer(
+    layer: string,
+    polygonFolder: string
+  ): { request: Promise<Object>; source: CancelTokenSource } {
+    const source = axios.CancelToken.source();
+    polygonFolder = "png_layers";
```
[the remark this answers is not in the document]
Comment (ManuelStardust) [reply src/utils/biabAPI.ts:61]: Solved

File: src/pages/Search.jsx (hunk 1 above)
Comment (ManuelStardust) [reply src/pages/Search.jsx]: I made an improvement please review.

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (ManuelStardust) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:66]: I couldn't understand how to solve this comment, @cazapatamar could you solve this and the previous one?

File: src/pages/Search.jsx (hunk 1 above)
Comment (erikasv) [reply src/pages/Search.jsx]: yeap, I think it's better

File: src/pages/Search.jsx [hunk 5]
```diff
@@ -599,7 +601,7 @@ class Search extends Component {
       styleName = ""
     } = options;
     const { selectedAreaId, selectedAreaTypeId } = this.props;
-    const { layers } = this.state;
+    const { layers, searchType } = this.state;
```
Comment (erikasv) [inline src/pages/Search.jsx:604]: this state is not being used in this function

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (erikasv) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:66]: @cazapatamar I think you should solve those issues in this same branch

File: src/pages/Search.jsx (hunk 5 above)
Comment (ManuelStardust) [reply src/pages/Search.jsx]: Solved

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (cazapatamar) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts]: Fixed!

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts
Comment (cazapatamar) [reply src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:1]: Fixed!

File: src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts [hunk 6]
```diff
@@ -56,72 +56,69 @@ export class ForestLossPersistenceController {
...
     latestPeriod: string,
     searchType: "definedArea" | "drawPolygon",
-    polygon: Polygon | null
+    polygon: Polygon | null,
+    polygonFolder: string
   ): Promise<ForestLPData> => {
     if (searchType === "drawPolygon") {
       return biabAPI
-        .requestForestLPData(polygon)
+        .requestForestLPData(polygon, polygonFolder)
```
Comment (erikasv) [inline src/pages/search/drawer/landscape/forest/ForestLossPersistenceController.ts:64]: Why did you remove `polygonFolder` prameter?
